File: core/src/main/java/at/shockbytes/corey/common/core/running/RunningManager.java
package at.shockbytes.corey.common.core.running;

import android.location.Location;

/**
 * @author Martin Macheiner
 *         Date: 05.09.2017.
 */

public interface RunningManager {

    void startRunRecording();

    void stopRunRecord(long timeInMs);

    Run updateCurrentRun(Location location);

    String getCurrentPace();

    Run getFinishedRun();

    boolean isRecording();

}

File: core/src/main/java/at/shockbytes/corey/common/core/running/RunningManager.kt
package at.shockbytes.corey.common.core.running

import at.shockbytes.corey.common.core.location.CoreyLocation

/**
 * @author Martin Macheiner
 * Date: 05.09.2017.
 */

interface RunningManager {

    val currentPace: String

    val finishedRun: Run

    val isRecording: Boolean

    fun startRunRecording()

    fun stopRunRecord(timeInMs: Long, userWeight: Double): Run

    fun updateCurrentRun(location: CoreyLocation): Run
}
